src/main/java/org/camunda/community/benchmarks/ProcessDeployer.java

```diff
@@ -43,29 +52,126 @@ public void autoDeploy() {
+            if (config.getJobTypesToReplace()!=null) {
+                // Split by "," if there are multiple task types to be replaced
+                String[] tasksToReplace = {config.getJobTypesToReplace()};
+                if (config.getJobTypesToReplace().contains(",")) {
+                    tasksToReplace = config.getJobTypesToReplace().split(",");
+                }
+                for (String taskToReplace: tasksToReplace) {
+                    fileContent = fileContent.replaceAll(taskToReplace, config.getJobType());
+                }
             }
-            for (String taskToReplace: tasksToReplace) {
-                fileContent = fileContent.replaceAll(taskToReplace, config.getJobType());
+            if (config.getBpmnProcessIdToReplace()!=null) {
+                fileContent = fileContent.replaceAll(config.getBpmnProcessIdToReplace(), config.getBpmnProcessId());
             }
         }
-        if (config.getBpmnProcessIdToReplace()!=null) {
-            fileContent = fileContent.replaceAll(config.getBpmnProcessIdToReplace(), config.getBpmnProcessId());
-        }
 
         return new ByteArrayInputStream(fileContent.getBytes());
     }
+
+    /**
+     * Inject unique job types for service tasks that don't have zeebe:taskDefinition
+     * Uses Zeebe's BPMN model API for robust and type-safe BPMN manipulation
+     */
+    private String injectUniqueJobTypes(String bpmnContent) throws Exception {
+        // First, check if zeebe namespace needs to be added
+        String modifiedContent = bpmnContent;
+        if (!bpmnContent.contains("xmlns:zeebe=")) {
+            // Add zeebe namespace declaration using string replacement for safety
+            modifiedContent = bpmnContent.replace(
+                "<bpmn:definitions",
+                "<bpmn:definitions xmlns:zeebe=\"http://camunda.org/schema/zeebe/1.0\""
+            );
+        }
+        
+        // Parse BPMN using Zeebe's BPMN model API
+        BpmnModelInstance modelInstance = Bpmn.readModelFromStream(new ByteArrayInputStream(modifiedContent.getBytes()));
+        
+        boolean modified = false;
+        
+        // Find all service tasks and inject job types where needed
+        Collection<ServiceTask> serviceTasks = modelInstance.getModelElementsByType(ServiceTask.class);
+        
+        for (ServiceTask serviceTask : serviceTasks) {
+            // Check if this service task already has a zeebe:taskDefinition
+            if (!hasZeebeTaskDefinition(serviceTask)) {
+                // Generate a unique job type based on the task ID
+                String taskId = serviceTask.getId();
+                String uniqueJobType = "benchmark-task-" + (taskId != null && !taskId.isEmpty() ? taskId : UUID.randomUUID().toString());
+                
+                // Create and add zeebe:taskDefinition using fluent approach
+                addZeebeTaskDefinition(serviceTask, modelInstance, uniqueJobType);
```

Comment:
Use the fluent API of the Zeebe BPMN Model API to modify the service task, e.g.
ServiceTaskBuilder builder = new ServiceTaskBuilder(modelInstance, serviceTask);
builder.zeebeJobType("your-job-type");

Then remove all the helper methods that you introduced for that.